sos/report/plugins/juju.py

```diff
@@ -186,6 +194,15 @@ def setup(self):
                         )
                         self.add_cmd_output(command, runas=juju_user)
 
+    def _juju_agent(self, command):
+        agent_name = ""
+        for dir_name in self.listdir("/var/lib/juju/agents"):
+            if re.search('machine-*|controller-*|application-*', dir_name):
+                agent_name = dir_name
+                break
+
+        return f"juju-introspect --agent={agent_name} {command}"
```

Comment:
Isnt it faster to store the `agent_name` in a variable and use it statically? Currently you call this method once for each `juju_agents_cmds` while you get a static info that can be fetched just once.

Or do I miss something?

Reply:
yup, good catch, was trying to mimic a different version of the same thing and didn't really think about it too much. I'll get that updated